ledger-core-ripple/src/ripple/keychains/RippleLikeKeychain.cpp

```diff
@@ -168,29 +168,39 @@ namespace ledger {
         RippleLikeKeychain::Address RippleLikeKeychain::derive() {
             if (_address.empty()) {
                 _localPath = getDerivationScheme()
-                        .setAccountIndex(getAccountIndex())
-                        .setCoinType(getCurrency().bip44CoinType)
-                        .setNode(0)
-                        .setAddressIndex(0).getPath().toString();
+                    .setCoinType(getCurrency().bip44CoinType)
+                    .getPath().toString();
 
                 auto cacheKey = fmt::format("path:{}", _localPath);
                 _address = getPreferences()->getString(cacheKey, "");
-
                 if (_address.empty()) {
-                    auto p = getDerivationScheme().getSchemeFrom(DerivationSchemeLevel::NODE).shift(1)
-                            .setAccountIndex(getAccountIndex())
-                            .setCoinType(getCurrency().bip44CoinType)
-                            .setNode(0)
-                            .setAddressIndex(0).getPath().toString();
-                    auto xpub = _xpub;
+                    auto nodeScheme = getDerivationScheme()
+                        .getSchemeFrom(DerivationSchemeLevel::NODE);
+                    auto p = nodeScheme.getPath().getDepth() > 0 ? nodeScheme
```

Comment:
maybe a comment will be appropriate here

Reply:
I agree 100%, but since I wasn’t the one writing the code, I have no idea what to write as comment. 😕